### PR TITLE
Add support for wodin

### PR DESCRIPTION
**Pull request by richfitz**
Adds the beginnings of functions to use from wodin

### CODE REVIEW

File: src/wodin.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    Individual = "Individual",
+    /** The mean over several Individual trace */
+    Mean = "Mean",
+    /** A single trace for where no variation was observed across all replicates (particles) */
+    Deterministic = "Deterministic"
+}
+
+/**
+ * Interface for objects returned by {@link wodinRunDiscrete}. This
+ * represents some number of model variables, each of which may
```
Comment (EmmaLRussell) [inline src/wodin.ts]: "each of which may..." what?

File: src/wodin.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+ */
+export interface DiscreteSeriesSet {
```
Comment (EmmaLRussell) [inline src/wodin.ts:37]: It feels a little bit unJSONy to have all these arrays which just happen to match each other by index. Given that we're going to have to reshape these anyway for plotly, would it be a little friendlier to have something like 
```
export interface DiscreteSeriesValues {
  name: string,
  y: number[],
  mode: DiscreteSeriesMode
}

export interface DiscreteSeriesSet {
  x: number
  values: DiscreteSeriesValues[]
}
``` 

I guess that would be inconsistent with what we get from the continuous models though, and maybe there's a reason why the current structure is easier to deal with?

File: src/wodin.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+ *
+ * @param pars Parameters to set into the model on construction
+ *
+ * @param timeStart The starting step (often 0)
+ *
+ * @param timeEnd The finishing step, greater than zero
+ *
+ * @param dt The size of each step
+ *
+ * @param nParticles The number of independent particles to run
```
Comment (EmmaLRussell) [inline src/wodin.ts]: This should be the number of variables?

File: src/wodin.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        const sol = tidyDiscreteSolution1(el.name, solution);
+        names.push(...sol.names);
+        y.push(...sol.y);
+        mode.push(...sol.mode);
+    });
+
+    return { mode, names, x, y };
+}
+
+export function tidyDiscreteSolution1(name: string, solution: DiscreteSolution) {
```
Comment (EmmaLRussell) [inline src/wodin.ts]: Could there be a better name for this? I got really confused when I hadn't clocked the `1` and thought `tidyDiscreteSolution` was recursively calling itself!

File: test/helpers.ts [hunk 5]
```diff
@@ -77,3 +77,7 @@ export function cumsum(x: number[]): number[] {
     })
     return ret;
 }
+
+export function rep(x: any, n: number): (typeof x)[] {
```
Comment (EmmaLRussell) [inline test/helpers.ts:81]: `rep` stands for 'repeat'?

**Review by EmmaLRussell — COMMENTED** [review]
I think I just about followed this!

File: src/wodin.ts (hunk 2 above)
Comment (richfitz) [reply src/wodin.ts:37]: I can do this, sure, no strong reason for the current format at all!

File: src/wodin.ts (hunk 3 above)
Comment (richfitz) [reply src/wodin.ts]: this is the number of replicate runs, it's unrelated to the number of variables

File: test/helpers.ts (hunk 5 above)
Comment (richfitz) [reply test/helpers.ts:81]: yeah, stolen from the R utility function of the same name (as with seq for sequence)

File: src/wodin.ts (hunk 3 above)
Comment (EmmaLRussell) [reply src/wodin.ts]: ah, of course!

File: src/wodin.ts (hunk 4 above)
Comment (richfitz) [reply src/wodin.ts]: sure, how about `tidyDiscreteSolutionVariable`?

**Review by EmmaLRussell — APPROVED** [review]
Looks good, though I'm still on tenterhooks as to "each of which may..." (l.20 wodin.ts)

**Comment by richfitz** [comment]
sorry about that, added now

**Comment by EmmaLRussell** [comment]
> sorry about that, added now

thanks, makes sense! :)

**Pull request merged**